### PR TITLE
New Modules: text2vec-aws and generative-aws

### PR DESCRIPTION
**Pull request by antas-marcin**
### What's being changed:

This PR introduces 2 new modules with integration to `AWS Bedrock` service:
- `text2vec-aws`
- `generative-aws`

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/generative-aws/clients/aws.go [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+
+		if v.isAmazonModel(model) {
+			body, err = json.Marshal(bedrockAmazonGenerateRequest{
+				InputText: prompt,
+				//TextGenerationConfig: &textGenerationConfig{
+				//	MaxTokenCount: *settings.MaxTokenCount(),
+				//	StopSequences: settings.StopSequences(),
+				//	Temperature:    *settings.Temperature(),
+				//	TopP:          int(*settings.TopP()),
+				//},
```
Comment (parkerduckworth) [inline modules/generative-aws/clients/aws.go:115]: Can these be removed? Or are they still needed?

File: modules/generative-aws/clients/aws.go [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+				res.StatusCode, *resBody.Message)
+		}
+		return nil, fmt.Errorf("connection to AWS Sagemaker failed with status: %d", res.StatusCode)
+	}
+
+	if len(resBody.Generations) == 0 {
+		return nil, fmt.Errorf("received empty response from AWS Sagemaker")
+	}
+
+	// todo fix this ID check, just a temp thing
```
Comment (parkerduckworth) [inline modules/generative-aws/clients/aws.go:276]: Is this still relevant?

File: modules/generative-aws/config/class_settings.go [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+	DefaultAnthropicTopK              = 250
+	DefaultAnthropicTopP              = 0.999
+)
+
+var DefaultAI21MaxTokens = 300
+
+// DefaultAnthropicStopSequences = []string{"\\n\\nHuman:"}
+// DefaultAnthropicTemperature   = 1.0
+// DefaultAnthropicTopK          = 250
+// DefaultAnthropicTopP          = 0.999
```
Comment (parkerduckworth) [inline modules/generative-aws/config/class_settings.go:60]: Still needed?

File: modules/generative-aws/config/class_settings.go [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+	// "sagemaker",
+}
+
+var availableBedrockModels = []string{
+	// Temporarily disable
+	// "anthropic.claude-instant-v1",
+	// "anthropic.claude-v1",
+	// "anthropic.claude-v2",
+	// "ai21.j2-mid-v1",
+	// "ai21.j2-ultra-v1",
```
Comment (parkerduckworth) [inline modules/generative-aws/config/class_settings.go:81]: Still need to be disabled?

File: modules/generative-aws/config/class_settings.go [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+			return &asInt
+		}
+		asNumber, ok := val.(json.Number)
+		if ok {
+			asFloat, _ := asNumber.Float64()
+			asInt := int(asFloat)
+			return &asInt
+		}
+		wrongVal := -1
+		return &wrongVal
```
Comment (parkerduckworth) [inline modules/generative-aws/config/class_settings.go:239]: `switch val.(type)` might work nicely here

**Review by parkerduckworth — APPROVED** [review]

File: modules/generative-aws/clients/aws.go (hunk 1 above)
Comment (antas-marcin) [reply modules/generative-aws/clients/aws.go:115]: removed

File: modules/generative-aws/clients/aws.go (hunk 2 above)
Comment (antas-marcin) [reply modules/generative-aws/clients/aws.go:276]: it's not, I have removed the comment

File: modules/generative-aws/config/class_settings.go (hunk 3 above)
Comment (antas-marcin) [reply modules/generative-aws/config/class_settings.go:60]: I have removed those settings I don't thing that it's needed. When I'll be re-writing this module I'll then re-introduce them if they will be still needed

File: modules/generative-aws/config/class_settings.go (hunk 4 above)
Comment (antas-marcin) [reply modules/generative-aws/config/class_settings.go:81]: I have removed commented part

File: modules/generative-aws/config/class_settings.go (hunk 5 above)
Comment (antas-marcin) [reply modules/generative-aws/config/class_settings.go:239]: I will keep that in my during my next iteration of fixes for this module, for now I will leave it as it.

**Review by aliszka — APPROVED** [review]

**Pull request merged**